app/commands/user/reputation_token/calculate_contextual_data.rb

```diff
@@ -0,0 +1,84 @@
+class User::ReputationToken
+  class CalculateContextualData
+    include Mandate
+    include ActionView::Helpers::TextHelper
+    include ActionView::Helpers::NumberHelper
+
+    def initialize(user_ids, period: nil, earned_since: nil, track_id: nil, category: nil)
+      @single_user = user_ids.is_a?(Integer)
+      @user_ids = Array(user_ids)
+      @track_id = track_id
+      @category = category
+
+      if earned_since
+        @earned_since = earned_since
+      elsif period
+        case period.to_sym
+        when :week
+          @earned_since = Time.zone.today - 6.days
```

Comment:
I wonder if doing it the same way it is done in the above class reads a tiny bit nicer:

```suggestion
          @earned_since = Time.zone.today - 1.week + 1.day
```

Reply:
I'm actually going to go the other way, purely because I can't actually use `1.month` as the value isn't 30, which is what I expected. But thank you for the nudge on consistency.